addons/l10n_it_stock_ddt/models/account_edi_format.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding:utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+
+
+class AccountEdiFormat(models.Model):
+    _inherit = 'account.edi.format'
+
+    def _l10n_it_invoice_is_direct(self, invoice):
+        """ An invoice is only direct if the Transport Documents are all done the same day as the invoice. """
+        if invoice.l10n_it_ddt_ids:
+            for ddt in invoice.l10n_it_ddt_ids:
+                if not ddt.date_done or ddt.date_done != invoice.invoice_date:
+                    return False
+        return True
+
+    def _l10n_it_get_invoice_features_for_document_type_selection(self, invoice):
+        res = super()._l10n_it_get_invoice_features_for_document_type_selection(invoice)
+        res['direct_invoice'] = self._l10n_it_invoice_is_direct(invoice)
+        return res
+
```

Comment:
Only TD24 should be the only one to collect deferred invoices (invoiced after the DDTs).
All other document types should collect just direct invoices (invoiced the same day as the DDT).